spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Optimizer.scala

```diff
@@ -232,6 +232,88 @@ abstract class Optimizer[T: ClassTag, D](
     this
   }
 
+  /**
+   * Set new train dataset
+   *
+   * @param sampleRDD training Samples
+   * @param batchSize mini batch size
+   * @return the Optimizer
+   */
+  def setTrainData(sampleRDD: RDD[Sample[T]],
```

Comment:
Shall we combine these methods, using None as default for optional arguments?

Reply:
I follow the same interface as the apply() function of optimizer.  Probabaly it's designed this way to avoid conditional clauses.

Reply:
changed both apply and setTrainData interface and merged several.